src/OnboardingSPA/components/GlobalStyleParent/index.js

```diff
@@ -0,0 +1,74 @@
+import { useSelect, useDispatch } from '@wordpress/data'; 
+import { useState, useEffect } from '@wordpress/element'; 
+
+import { store as nfdOnboardingStore } from '../../store';
+import { getGlobalStyles, setGlobalStyles } from '../../utils/api/themes';
+import { useGlobalStylesOutput } from '../../utils/global-styles/use-global-styles-output';
+
+/**
+ * Global Style Parent Component
+ * The Fetching of Global Style Object from either store or API is 
+ * common to a lot many places and this component does the trick import { useState, useEffect } from '@wordpress/element';for us.
+ *
+ * @returns Global Style Parent
+ */
+
+const GlobalStyleParent = ({ children }) => {
```

Comment:
```suggestion
const GlobalStylesProvider = ({ children }) => {
```
And this component needs to go inside the `LivePreview` components so it is easier to migrate to a new module in the future.

Reply:
Won't that increase our work as then every live preview component is gonna do this computation that is unnecessary as if this is a parent then it will only be done once.